### ISSUE TITLE
JWT 

### ISSUE THREAD

**Issue by iloop2020** [GoogleCloudPlatform/iot-device-sdk-embedded-c#127]
Hi 

We foudn the MQTT will be disconnect to cloud about 1hour long, in iot_core_mqtt_client.c example
there is a parameter 3600 Second, the 1hour limitation is controlled by that parameter ?

`  iotc_state_t state = iotc_create_iotcore_jwt(
      iotc_project_id,
      /*jwt_expiration_period_sec=*/3600, &iotc_connect_private_key_data, jwt,
      IOTC_JWT_SIZE, &bytes_written);`
      
      
If yes, what is the Max value for that parameter ?
Thank you very much .

**Comment by gguuss** [comment]
The 3600 second parameter is for the JWT expiration. The maximum valid duration for a JWT for Cloud IoT Core is 24 hours or 86,400 seconds.  If you're getting a disconnect and it is not successfully re-connecting, that may indicate something else is wrong in the library but I'm not actively maintaining this project currently.

**Comment by iloop2020** [comment]
Hi @gguuss 

Thank you for your support.

With 3600 second parameter, does it expected to be disconnected after 1hour ?

Thank you.

**Comment by atigyi** [comment]
Yes, with jwt expiration set to 3600, device is expected to be disconnected after 1 hour. See e.g. [Refreshing JWTs](https://cloud.google.com/iot/docs/how-tos/credentials/jwts#refreshing_jwts).
